### PR TITLE
refactor(rust,python): refactor `arange` and add `int_range`/`int_ranges`

### PR DESCRIPTION
**Pull request by stinodego**
Partially addresses #9019

Changes:
* Refactor `arange` to make use of the existing expression architecture
* Add `int_range` and `int_ranges` functions as alias for `arange` functionality.
  * While `arange` has some merit naming-wise (exists in `numpy`, for example), it's not a good name. What exactly is an 'array range'? We have `time_range`, `date_range`, let's be explicit and call this what it is: an `int_range`. We could add a `float_range` later, for example.
  * Splits out the vertical vs horizontal behaviours
* Add doc entry to Expressions / Functions

Future steps (not included here to keep diff somewhat small):
* Further refactor the functions, they're still messy.
* Support a `dtype` arg for `int_range(s)` on the Rust side to avoid materializing everything as `Int64` first.
* Rename the feature?
* Deprecate `arange`?

Can we deprecate `arange` as a whole, or do we want to keep this function due to name familiarity? If we keep it, do we want to split out functionality into  `arange`/`aranges`? I prefer getting rid of it entirely.

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Nice!

Is the idea to introduce `int_range` / `int_ranges` now, deprecate `arange` in favour of them, and then in 1.0 remove `arange` completely?

---

Just a suggestion testing-wise - perhaps it'd be good to test the schema in the lazy case, like how is done in the `test_date_range_schema`, / `test_date_range_no_alias_schema_9037` tests?

**Comment by stinodego** [comment]
> Is the idea to introduce `int_range` / `int_ranges` now, deprecate `arange` in favour of them, and then in 1.0 remove `arange` completely?

That's what I would propose. But I can imagine Ritchie might want to keep `arange` for its name familiarity. Either way, this PR keeps existing `arange` functionality, so we can do the deprecation later if we choose to.

> Just a suggestion testing-wise - perhaps it'd be good to test the schema in the lazy case, like how is done in the `test_date_range_schema`, / `test_date_range_no_alias_schema_9037` tests?

Thanks, I'll look into it!

File: py-polars/polars/functions/range.py [hunk 1]
```diff
@@ -139,6 +145,177 @@ def arange(
...
+    end: IntoExpr,
+    step: int = ...,
+    *,
+    dtype: PolarsIntegerType = ...,
+    eager: bool,
+) -> Expr | Series:
+    ...
+
+
+def int_ranges(
```
Comment (ritchie46) [inline py-polars/polars/functions/range.py:265]: So the return type of this is always  a list? Contrary to default `int_range` which will be an int type?

**Review by ritchie46 — APPROVED** [review]
Looks good. I like that we moved the implementation properly to the new architecture. 

On the naming of `arange`. I think we should keep it around as an alias to `int_range`. I believe the range in almost all programming language defaulted to integers. But given that we clarify the `dtype` for `time` and `date`, I understand the specific naming need.

File: py-polars/polars/functions/range.py (hunk 1 above)
Comment (stinodego) [reply py-polars/polars/functions/range.py:265]: Exactly! That is the intention of this split. We will do the same for the other ranges  (`date_range`, `time_range`).

**Comment by stinodego** [comment]
> I think we should keep it around as an alias to `int_range`.

Sounds fair enough. Then I'll deprecate the part of `arange` that dispatches to `int_ranges`.

**Comment by Bidek56** [comment]
Is `int_range` and `int_ranges` functions only available in `py-polars`?
I am trying to upgrade `nodejs-polars` to `rs-0.31.1` but I am getting:
`error[E0425]: cannot find function 'int_ranges' in module 'dsl'` Thx

**Comment by stinodego** [comment]
It should be available in Rust!

**Comment by stinodego** [comment]
Ah yes. The feature has been renamed from `arange` to `range`. So you must enable the correct feature for it to be available.

**Comment by Bidek56** [comment]
Adding: `#[cfg(feature = "range")]` decorator solves it. Thanks!!

**Comment by Bidek56** [comment]
I am still getting `not yet implemented` when trying to run a [test](https://github.com/Bidek56/nodejs-polars/blob/20574a4d0b2fa27b159ff179d536a2439f9692e4/__tests__/lazy_functions.test.ts#L131), but that could some other issue.

**Comment by ritchie46** [comment]
> Adding: `#[cfg(feature = "range")]` decorator solves it. Thanks!!

Are you sure? You should activate it on your cargo.toml. This decorator will only ensure your code is excluded from compilation until you activate that feature on your crate.

**Comment by Bidek56** [comment]
You are correct, `#[cfg(feature = "range")]` is not needed.

Adding these lines to [Cargo.toml](https://github.com/pola-rs/nodejs-polars/pull/85/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542) solves it, is this the best approach? Thx
```
[features]
default = ["range"]
range = ["polars-lazy/range"]
```

**Comment by ritchie46** [comment]
Yes.

**Pull request merged**